main.tf

```diff
@@ -55,7 +65,7 @@ resource "aws_redshift_parameter_group" "this" {
 }
 
 resource "aws_redshift_subnet_group" "this" {
-  count = "${local.enable_create_redshift_subnet_group}"
+  count = "${var.redshift_subnet_group_name != "" ? 0 : 1}"
```

Comment:
It should be `==` instead of `!=`

Reply:
Come to think of it, should we change the `locals` to be similar to the Terraform RDS module? https://github.com/terraform-aws-modules/terraform-aws-rds/blob/master/main.tf#L1-L6

Reply:
Yes, using `locals` is much better.

Reply:
The way it was implemented before, using the blocks:
```
  enable_create_redshift_subnet_group    = "${var.redshift_subnet_group_name == "" ? 0 : 1}"
  enable_create_redshift_parameter_group = "${var.parameter_group_name == "" ? 0 : 1}"
```

Was actually was caused the problem. If you look at my logs in https://github.com/terraform-aws-modules/terraform-aws-redshift/issues/4, for some reason, using `enable_create_redshift_` the way it was written resulted in subnet groups being created even when a user specified an already existing subnet group.

In the RDS module, the difference is that the code block:

```
  enable_create_db_subnet_group    = "${var.db_subnet_group_name == "" ? var.create_db_subnet_group : 0}"
  enable_create_db_parameter_group = "${var.parameter_group_name == "" ? var.create_db_parameter_group : 0}"
```

Has explicit `create_db_*` variables. Do we want equivalent `create_redshift_parameter_group` and `create_redshift_subnet_group` variables in this module?